### PR TITLE
feat(richtext): add query APIs to AttributeContainer

### PR DESCRIPTION
**Pull request by mkeeda**
## Overview
Adds helpful query and introspection APIs to `AttributeContainer`. This allows for more intuitive and concise condition checks within the `runs(predicate)` closures, eliminating the need to compare `getOrNull(key)` results against `null`.

## Implementation Details
- Added the following APIs to `AttributeContainer`:
  - `isNotEmpty(): Boolean`
  - `size: Int`
  - `keys: Set<AttributeKey<*>>`
  - `containsKey(key: AttributeKey<*>): Boolean`
  - `containsAll(vararg keys: AttributeKey<*>): Boolean`
  - `containsAny(vararg keys: AttributeKey<*>): Boolean`
- Added comprehensive tests for all new APIs in `AttributeContainerTest.kt`.

### CODE REVIEW

File: richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/AttributeContainer.kt [hunk 1]
```diff
@@ -33,6 +33,38 @@ public class AttributeContainer private constructor(
...
+
+    /**
+     * Returns true if this container contains the specified [key].
+     */
+    public fun containsKey(key: AttributeKey<*>): Boolean = attributes.containsKey(key)
+
+    /**
+     * Returns true if this container contains all of the specified [keys].
+     */
+    public fun containsAll(vararg keys: AttributeKey<*>): Boolean = keys.all { containsKey(it) }
```
Comment (mkeeda) [inline richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/AttributeContainer.kt:62]: The `containsAll` and `containsAny` functions have notable edge-case behavior when called with no arguments (empty `vararg`):

- `containsAll()` → returns `true` (vacuous truth: `all {}` on an empty collection is always true)
- `containsAny()` → returns `false` (`any {}` on an empty collection is always false)

These may be surprising to callers. Consider one of the following:
1. Documenting this behavior explicitly in the KDoc (e.g., `"Returns true if no keys are specified (vacuous truth)."`).
2. Adding test cases to define and guard this behavior:

```kotlin
@Test
fun `containsAll returns true when no keys are specified`() {
    val container = attributeContainerOf()
    container.containsAll() shouldBe true
}

@Test
fun `containsAny returns false when no keys are specified`() {
    val container = attributeContainerOf()
    container.containsAny() shouldBe false
}
```

File: richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/AttributeContainer.kt [hunk 2]
```diff
@@ -33,6 +33,38 @@ public class AttributeContainer private constructor(
...
+     * Returns the number of attributes stored in this container.
+     */
+    public val size: Int
+        get() = attributes.size
+
+    /**
+     * Returns the set of attribute keys currently stored in this container.
+     */
+    public val keys: Set<AttributeKey<*>>
+        get() = attributes.keys
```
Comment (mkeeda) [inline richtext/src/main/kotlin/dev/mkeeda/arranger/richtext/AttributeContainer.kt]: `attributes.keys` returns the key set of the internal `Map`, which is a live view into the map's key entries. While `attributes` is typed as `Map<AttributeKey<*>, Any?>` (not `MutableMap`), a defensive copy via `toSet()` would make the immutability contract of `AttributeContainer` more explicit and safer against future internal changes.

```kotlin
// Suggestion: return an immutable snapshot
public val keys: Set<AttributeKey<*>>
    get() = attributes.keys.toSet()
```

This is a low-severity suggestion — no actual bug exists here — but it aligns well with the immutable design philosophy of `AttributeContainer`.

**Review by mkeeda — COMMENTED** [review]
(AI review)

The implementation is clean and consistent with the existing API design of `AttributeContainer`. All new methods are properly documented with KDoc, and the test coverage is comprehensive across the positive and negative paths.

Two minor points noted via inline comments:
1. **`containsAll` / `containsAny` with empty `vararg`** — the edge-case behavior (vacuous truth / always-false) should be documented or covered by tests.
2. **`keys` property** — consider a defensive `toSet()` copy to reinforce the immutability contract, though no actual bug exists currently.
